XenAdmin/SettingsPanels/GpuEditPage.cs

```diff
@@ -338,10 +343,10 @@ public VGpuDetailRow(VGPU vGpu)
 
         private void SetCells()
         {
-            if (int.TryParse(VGpu.device, out var device) && device != 0)
-                deviceColumn.Value = device;
+            if (int.TryParse(VGpu.device, out var device))
+                deviceColumn.Value = device + VGPU_PCI_SLOT_NUMBER_SHIFT;

```

Comment:
For older versions I don't think it matters because the code populating the GridView further up hides the column, but if we want to be consistent we may want to write:
```
if (int.TryParse(VGpu.device, out var device) &&
    (Helpers.PlymouthOrGreater(VGpu.Connection) || device != 0))
    deviceColumn.Value = device + VGPU_PCI_SLOT_NUMBER_SHIFT;
else
    deviceColumn.Value = Messages.HYPHEN;
```

Reply:
But for new versions, in case the device nubmer used [0,1,2] will be shown as [-,12,13], just to confirm, does that make sense?

Reply:
but the `PlymouthOrGreater` judge is a good point, perhaps we can write someting like
```if (int.TryParse(VGpu.device, out var device) )
    deviceColumn.Value = Helpers.PlymouthOrGreater(VGpu.Connection)? device + VGPU_PCI_SLOT_NUMBER_SHIFT: device;
else
    deviceColumn.Value = Messages.HYPHEN;
```